### PR TITLE
Spinner for remote data provider

### PR DESCRIPTION
**Pull request by Toshegg**
Depends on https://github.com/vaadin/vaadin-grid/pull/1114

### CODE REVIEW

File: px-data-grid.html [hunk 1]
```diff
@@ -239,6 +263,11 @@
              */
             remoteDataProvider: {
               type: Function
+            },
+
+            loading: {
```
Comment (limonte) [inline px-data-grid.html]: - documentation is missing
- should be read-only
- test is missing

Comment (Toshegg) [reply px-data-grid.html]: Done.

**Comment by limonte** [comment]
- tests are failing, even after https://github.com/vaadin/vaadin-grid/pull/1114 merged
- needs rebase

File: bower.json [hunk 2]
```diff
@@ -18,7 +18,8 @@
     "iron-scroll-target-behavior": "^v2.0.0",
     "px-icon-set": "^2.0.0",
     "px-dropdown": "^4.0.0",
-    "vaadin-grid": "^4.0.0"
+    "vaadin-grid": "master",
```
Comment (limonte) [inline bower.json]: revert this line, https://github.com/vaadin/vaadin-grid/releases/tag/v4.1.0-alpha5 was just released

Comment (Toshegg) [reply bower.json]: Done.

**Comment by limonte** [comment]
Doesn't work properly in this case:

recorded

**Comment by Toshegg** [comment]
@limonte, done.

File: px-data-grid.html [hunk 3]
```diff
@@ -366,6 +403,16 @@
             this._updateActionMenu();
           });
         }
+
+        _loadingChanged(loading) {
+          clearTimeout(this._spinnerHiddenTimeout);
+
+          if (loading) {
+            this._spinnerHiddenTimeout = setTimeout(() => this._spinnerHidden = false, 500);
```
Comment (limonte) [inline px-data-grid.html]: `500` looks like a magic number here. should we provide the ability to change the debounce timeout? Something like

```js
loadingSpinnerDebounce: {
  type: Number,
  value: 500
}

File: test/px-data-grid-test.js [hunk 4]
```diff
@@ -37,12 +37,33 @@ function runTests() {
...
+        expect(grid._spinnerHiddenTimeout).to.be.ok;
+      });
+
+      it('should hide spinner unless 500ms have passed', () => {
+        grid.remoteDataProvider = (params, callback) => {};
+        expect(grid.shadowRoot.querySelector('px-spinner').hasAttribute('hidden')).to.be.true;
+      });
+
+      it('should show spinner after 500ms when data is loading', () => {
+        const spy = sinon.spy(window, 'setTimeout');
```
Comment (limonte) [inline test/px-data-grid-test.js]: this test is hard to read and maintain, can it be simplified?

File: px-data-grid.html (hunk 3 above)
Comment (Toshegg) [reply px-data-grid.html]: Done.

File: test/px-data-grid-test.js (hunk 4 above)
Comment (Toshegg) [reply test/px-data-grid-test.js]: Done.

**Pull request merged**